n_plus_one_example/src/db.rs

```diff
@@ -0,0 +1,62 @@
+use dotenv::dotenv;
+use std::env;
+use tokio_postgres::{Error, NoTls};
+
+pub async fn connect_db() -> Result<
+    (
+        tokio_postgres::Client,
+        tokio::task::JoinHandle<Result<(), Error>>,
+    ),
+    Error,
+> {
+    dotenv().ok();
+
+    // Get environment variables
+    let username = env::var("username").expect("username not set in .env");
+    let password = env::var("password").expect("password not set in .env");
+    let dbname = env::var("dbname").expect("dbname not set in .env");
+
+    let (client, connection) = tokio_postgres::connect(
+        &format!(
+            "host=localhost user={} password={} dbname={}",
+            username, password, dbname
+        ),
+        NoTls,
+    )
+    .await?;
+
+    let handle = tokio::spawn(async move {
+        if let Err(e) = connection.await {
```

Comment:
We could add `connection.await.unwrap()`. While this will cause the program to panic if the connection is lost. The program will pass `cargo clippy` and `cargo build`!